ssaHAL/ssa/core.py

```
@@ -138,29 +162,80 @@ def __mqtt_connect(self,
+            print("[WARNING Received message from invalid topic. Ignoring.")
             return
 
-        print(f"[DEBUG] Executing action callback for {action}")
+        if action in self.__action_cb_dict:
+            func = self.__action_cb_dict[action].callback
+            try:
+                func(self, msg)
+            except Exception as e:
+                print(f"[ERROR] Action callback `{func.__name__}` failed: {e}")
+            finally:
+                return
```

Ruff (0.8.2)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)